metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -140,39 +191,61 @@ final class ReferenceProvider(
             definition.positionOccurrences(source, params.getPosition, doc)
           if (posOccurrences.isEmpty)
             // handling case `import a.{A as @@B}`
-            occerencesForRenamedImport(source, params, doc)
+            occurrencesForRenamedImport(source, params, doc)
           else posOccurrences
         }
         if (results.isEmpty) {
           scribe.debug(
             s"No symbol found at ${params.getPosition()} for $source"
           )
         }
-        results.map { result =>
-          val occurrence = result.occurrence.get
-          val distance = result.distance
-          val alternatives =
-            referenceAlternatives(occurrence.symbol, source, doc)
-          val locations = references(
+        val semanticdbResult = Future.sequence {
+          results.map { result =>
+            val occurrence = result.occurrence.get
+            val distance = result.distance
+            val alternatives =
+              referenceAlternatives(occurrence.symbol, source, doc)
+            val locations = references(
+              source,
+              params,
+              doc,
+              distance,
+              occurrence,
+              alternatives,
+              params.getContext.isIncludeDeclaration,
+              findRealRange,
+              includeSynthetics,
+            )
+            locations.map(ReferencesResult(occurrence.symbol, _))
+          }
+        }
+        val pcResult =
+          pcReferences(
             source,
             params,
-            doc,
-            distance,
-            occurrence,
-            alternatives,
-            params.getContext.isIncludeDeclaration,
-            findRealRange,
-            includeSynthetics,
+            forceIncludeLocalSearch = false,
+            path => !index.contains(path.toNIO),
           )
-          ReferencesResult(occurrence.symbol, locations)
-        }
-      case None => Nil
+
+        Future
+          .sequence(List(semanticdbResult, pcResult))
```

Comment:
If we have working semanticdb, we should not use pcResult, no? Or just as a fallback or when the symbol is `local123`.

Will this not cause us to always calculate both?

Reply:
When searching for `pcResult`, we only search the files that have occurrences of a particular token and sementicdb is missing for them. So if semanticdb is working and present for all the files it should really use `pcResult`.